app/src/pages/review/[resourceType]/[id].tsx

```diff
@@ -34,16 +140,149 @@ export default function CommentPage() {
+                              </List.Item>
+                            </List>
+                          </HoverCard.Dropdown>
+                        </HoverCard>
+                      </Group>
+                    }
+                    icon={<Star opacity={0.5} />}
+                    placeholder="Type"
+                    data={[
+                      { value: 'documentation', label: 'documentation' },
+                      { value: 'guidance', label: 'guidance' },
+                      { value: 'review', label: 'review' }
+                    ]}
+                    {...form.getInputProps('type')}
+                  />
+                </Group>
+                <Textarea
+                  radius="lg"
+                  mt="md"
+                  minRows={15}
+                  maxRows={15}
+                  placeholder="Your Artifact comment"
+                  label="Artifact Comment"
+                  description="Add a comment to the artifact"
+                  withAsterisk
+                  {...form.getInputProps('comment')}
+                />
+                <Space h="md" />
+                <Group grow>
+                  <TextInput radius="lg" label="Endorser Name" placeholder="Name" {...form.getInputProps('name')} />
+                  <Checkbox
+                    ref={ref}
+                    id="checkbox"
+                    color="white"
+                    mt="md"
+                    label="Include Date in Comment"
+                    {...form.getInputProps('date')}
+                    onChange={() => {
+                      setDateSelected(!dateSelected);
+                    }}
+                  />
+                  <Space h="md" />
+                </Group>
+                <Space h="md" />
+                <Group position="right" mt="md">
+                  <Button
+                    loading={isLoading}
+                    type="submit"
+                    onClick={() => {
+                      if (form.isValid()) {
+                        setIsLoading(true);
+                        setTimeout(() => {
+                          setDateSelected(false);
+                          form.reset();
+                          if (ref?.current?.checked) {
+                            ref.current.checked = false;
+                          }
+                          setIsLoading(false);
+                        }, 1000);
+                        const [additions, deletions] = parseUpdate(
+                          form.values.comment,
+                          form.values.type,
+                          form.values.name,
+                          dateSelected
+                        );
+                        resourceUpdate.mutate({
+                          resourceType: resourceType as ArtifactResourceType,
+                          id: resourceID as string,
+                          additions: additions,
+                          deletions: deletions
+                        });
```

Comment:
I would put these in an `if` statement or otherwise something that ensures that they are only called when `authoring === true`. Right now we only want to support adding the comment on the JSON for draft artifacts, but by calling `parseUpdate` here, the `resource.extension` is being temporarily set in the JSON but not saved in it (which may be confusing for the user). Additionally, the `mutate` procedure that gets called which we actually don't want. It still gets called and is successful because it tries to add additional information to a collection in the draft database but with an `id` that doesn't exist (since it's getting the id from the artifact in the central measure repository). `collection.insertOne()` is still successful, but it doesn't add anything so the pop up is misleading. Let me know if you want to talk through this more!